src/plugins/controls/public/react_controls/control_group/types.ts

```diff
@@ -66,7 +69,9 @@ export type ControlGroupApi = PresentationContainer &
     ignoreParentSettings$: PublishingSubject<ParentIgnoreSettings | undefined>;
     allowExpensiveQueries$: PublishingSubject<boolean>;
     untilInitialized: () => Promise<void>;
-    openAddDataControlFlyout: () => void;
+    openAddDataControlFlyout: (settings?: {
```

Comment:
Since settings is just a wrapper around controlInputTransform? How about just `openAddDataControlFlyout: (controlInputTransform?: ControlInputTransform)`

Reply:
There used to be more than one setting for the legacy (`onSave`) - I still haven't fully narrowed down which ones are necessary, and I won't know until the control group renderer is fully converted. For now, maybe we keep that extra level?